src/librustc_mir/build/expr/into.rs

```diff
@@ -305,9 +303,12 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                             -> BlockAnd<()>
         where F: FnOnce(&LoopScope) -> BasicBlock
     {
-        let loop_scope = self.find_loop_scope(span, label);
-        let exit_block = exit_selector(&loop_scope);
-        self.exit_scope(span, loop_scope.extent, block, exit_block);
+        let (exit_block, extent) = {
+            let loop_scope = self.find_loop_scope(span, label);
+            loop_scope.might_terminate = true;
```

Comment:
@nagisa seems like we should toggle this flag in `exit_scope`, not here, and then return the result (as I wrote elsewhere)